src/main/java/com/eternalcode/core/chat/ChatManager.java

```diff
@@ -0,0 +1,99 @@
+package com.eternalcode.core.chat;
+
+import com.eternalcode.core.configuration.ConfigurationManager;
+import com.eternalcode.core.configuration.MessagesConfiguration;
+import com.eternalcode.core.configuration.PluginConfiguration;
+import com.eternalcode.core.utils.ChatUtils;
+import com.eternalcode.core.utils.DateUtils;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.papermc.paper.event.player.AsyncChatEvent;
+import lombok.Getter;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import panda.std.Option;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public final class ChatManager {
+
+    @Getter private final Cache<UUID, Long> chatCache;
+    @Getter private boolean                 enabled;
+    private final char[]                    clearMessage;
+    private final ConfigurationManager      configManager;
+
+    public ChatManager(ConfigurationManager manager){
+        this.configManager = manager;
+
+        this.chatCache = CacheBuilder.newBuilder()
+            .expireAfterWrite(configManager.getPluginConfiguration().chatSlowMode * 1000L, TimeUnit.SECONDS)
+            .build();
+        this.enabled = true;
+        this.clearMessage = new char[30000];
+        Arrays.fill(this.clearMessage, ' ');
```

Comment:
Czy nie wystarczy w minecraft użycie \n? 30000 charów to trochę dużo

Reply:
Nie jestem pewny, zawsze robiłem tak. Przetestuję to